python/GangaDirac/Lib/Files/DiracFile.py

```diff
@@ -137,7 +137,7 @@ def locations(self):
 
     def _setLFNnamePattern(self, lfn="", namePattern=""):
 
-        if self.defaultSE != "":
+        if hasattr(self, 'defaultSE') and self.defaultSE != "":
```

Comment:
No, it's not there when you're reading in from an XML where the schema hasn't been written with the `defaultSE` as the first attribute within the file. Also this could happen during object construction as the Schema is _not_ an ordered dict and so the order in which the parameters arrive here is effectively random (we're just lucky that in 99% of cases it's sequential)